018-cli-2.0/018-cli-2.0.md

```diff
@@ -0,0 +1,312 @@
+
+    ```sh
+    $ appwrite login
+    Email:
+    Password:
+    ```
+
+    In CI environments, the authentication will continue to take place using the API key on a per project basis until we develop a new token / key based mechanism for console authentication. 
+
+    ```sh
+    appwrite client --setKey="" --setEndpoint="" ...
+    ```
+
+* Second method relies on a browser based auth.
+
+  ```sh
+  $ appwrite login
+
+  # Browser based authentication continues. After a successful login, the returned token and user ID are stored in a hidden preferences file.
+  ```
+
+  1. The command first finds an available local port. 
+  2. Construct a redirect URL using the available port `https://localhost:1234`
+  3. Spin up a local http server listening to on the available port ( 1234 in this case )
+  4. Opens the browser with a request to either the local appwrite server or appwrite cloud `/login?callback=https://localhost:1234` with the redirect URL as one of the query parameters.
+
+  ![CLI Auth](cli_auth.png)
+
+  This approach may seem unnecessarily complex now since we only allow email password based logins in the console. But as we move towards supporting OAuth logins in the console, this is the approach that needs to be followed. [Firebase CLI](https://github.com/firebase/firebase-tools/blob/0b0459bfde3fabf41223bb6d5d39a5cf325f1010/src/auth.ts#L476) for reference. 
+
+
+## Easier Deployment of cloud functions
+--- 
+
+### 🟢 Constrain deployments from current directory.
+
+We start by removing the `--code` parameter in the `createTag` command and constrain function deployments only from the current directory. 
+
+The refactored command will look like this 
+
+```sh
+appwrite functions createTag --functionId=[ID] --command=[COMMAND]
+```
+
+This corresponding appwrite issue will also need to be tackled as we implement this.
+https://github.com/appwrite/appwrite/issues/1316
+
+### 🟢 The `.appwrite.json` File
+
+The next improvement is going to be along the lines of a `.appwrite.json` file. Running `appwrite init` in the current directory will create a `.appwrite.json` file and initialize the current directory with attributes necessary to communicate with a particular project.
+
+Consider this example
+```sh
+mkdir temp && cd temp
+
+# Initialize the current directory with an Appwrite project 
+appwrite init
+
+temp 
+└── .appwrite.json
+```
+
+A `.appwrite.json` file will be associated with one project only whose ID will be visible in the `project` attribute. 
+
+```json
+{
+    "project" : "",
+    "functions": [
+        {
+          "name" : "My awesome function 💪",
+          "path" : "./FunctionOne",
+          "command" : "python main.py",
+          "runtime": "python3.9",
+          // This section is useful when we allow cloud functions to be tested locally
+          "environment_variables" :{
```

Comment:
```suggestion
          "vars" :{
```

This is the name used here: https://appwrite.io/docs/server/functions#functionsCreate